common/src/deposits.rs

```diff
@@ -0,0 +1,114 @@
+//! This is the transaction analysis module
+//!
+
+use bitcoin::opcodes::all as opcodes;
+use bitcoin::script::PushBytesBuf;
+use bitcoin::taproot::LeafVersion;
+use bitcoin::taproot::NodeInfo;
+use bitcoin::taproot::TaprootSpendInfo;
+use bitcoin::Address;
+use bitcoin::Network;
+use bitcoin::ScriptBuf;
+use bitcoin::XOnlyPublicKey;
+use clarity::codec::StacksMessageCodec;
+use clarity::vm::types::PrincipalData;
+use secp256k1::SECP256K1;
+
+/// This struct contains the key variable inputs when constructing a
+/// deposit script address.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DepositScript {
```

Comment:
Can you update this to also "extract" the block height at which the reclaim script becomes spendable?

Also, if I understand this correctly, the depositor provides all the parameters including block height at which they can redeem and then here we're making the taproot script that we'd expect given those parameters and will use that to check whether the lock script does what's expected?

Reply:
I think getting the block height might be trickier than it seems. What happens if the deposit request exists on multiple forks with different heights? The block height is not a property of the request, but rather something we can derive from the chain state when we observe a request on the chain.

From that perspective, when it comes to the block height I think it makes sense to query that information from the database when we need it.

Reply:
Oh right forking! I knew it wasn't a property of the request since it's a relative time lock, but yeah, gotta handle forking. In any case, I was interpreting the "block height" question as "when are we going to extract that parameter in the reclaim script?" which is basically all that I have planned for in this library with respect to the reclaim script.

Also, if forking weren't bad enough the parameter that is fed into `OP_CHECKSEQUENCEVERIFY` can be either a relative time or the number of bitcoin blocks (I know you know this already). The bad part is that the choice happens when you spend the UTXO (as opposed to when you create the UTXO). That distinction was lost on me when I was first reading up about it. This means that now we have to care about time as well.